scripts/fuzz_opt.py

```diff
@@ -275,12 +275,28 @@ def compare(x, y, context):
         ))
 
 
+# converts a possibly-signed integer to an unsigned integer
+def unsign(x, bits):
+    return x & (2**bits - 1)
```

Comment:
- How does &ing with 111...11 make `x` unsigned? Maybe I'm missing something
- If it is only used in `to_64_bit`, which is an inner function within `numbers_are_close_enough`, why is it not an inner function too? Is it intended to be used elsewhere as well?

Reply:
It's a simple bitwise trick to convert signed to unsigned. For example `-1 & (2**8 - 1)  == 255`, etc. It works because the signed and the unsigned representations always differ by `2**bits`, or by nothing, in twos complement. So we can write `S = U + x`, where x is either `2**bits` or 0. And then if we apply `& (2**bits - 1)` to both sides,
```
S & (2**bits - 1) = (U + x) & (2**bits - 1) = U
```
(because masking U with `2**bits - 1` has no effect, and x has no bits that overlap with the mask).

I guess the function could be an inner function. I don't feel strongly. Would you prefer it nested in the function, or even nested further into the branch of code that uses it?

Reply:
Sorry for the basic question, but I realized I don't understand more basic things here:
- Why do we need to 'unsign' here?
- Is this 'unsign' something like converting -5 to 5?

Reply:
Oh, the issue is that the same bits can be reported as either signed or unsigned. `11111111` is 255 as unsigned and -1 as signed, and as they represent the same `i8`, we want to see that they are the same (not `i8` because there isn't such a wasm type, but the same for `i32`). The concrete issue here is that the same bits can be reported in different ways by the wasm2c fuzzer, and by other fuzzers.

In theory we should have a single uniform way to report numbers (say, always as unsigned). But that's hard because of JS legalization, which we also fuzz, which makes us sometimes report an i64 as two i32s, etc. This PR handles both legalization and signedness by always converting to the same format before comparing.

Reply:
Sorry for possibly more stupid question 😅
`numbers_are_close_enough` takes two strings and converts them into integers using `int` and here we `unsign` them. Then assuming we use `i8` instead of `i32` for illustration purpose, if we are given '-3' and '5' in `numbers_are_close_enough`, after `unsign` them, they become the same number and `numbers_are_close_enough` will return true. Is this correct and intended? If so, why? Aren't they different numbers?

About function location I'm not really opinionated; I was just wondering why one is inside and the other is outside.

Reply:
Good question, yes, in general it would not be correct to say `-3` and `5` are identical. However, if we know that they are `i8` values, then it would be correct. That is, in that type, those must represent identical bits. And the code path changed in this PR is one in which we know the value is an `i64`.

Reply:
Thanks for the patient explanation!